app/scripts/controllers/network/provider-api-tests/helpers.js

```diff
@@ -275,25 +282,34 @@ export async function withMockedInfuraCommunications(...args) {
 }
 
 /**
- * Builds a provider from the Infura middleware along with a block tracker, runs
- * the given function with those two things, and then ensures the block tracker
- * is stopped at the end.
+ * Builds a provider from the middleware (for the provider type) along with a
+ * block tracker, runs the given function with those two things, and then
+ * ensures the block tracker is stopped at the end.
  *
- * @param {WithInfuraClientArgs} args - Either an options bag + a function, or
- * just a function. The options bag, at the moment, may contain `network` (that
- * is, the Infura network; defaults to "mainnet"). The function is called with
- * an object that allows you to interact with the client via a couple of methods
- * on that object.
+ * @param {WithClientArgs} args - Either an options bag + a function, or
+ * just a function. The options bag, at the moment, may contain `network`
+ * (defaults to "mainnet"). The options bag may also include providerType
+ * The function is called with an object that allows
+ * you to interact with the client via a couple of methods on that object.
  * @returns {Promise<any>} The return value of the given function.
  */
-export async function withInfuraClient(...args) {
```

Comment:
Nit: I don't think it makes sense anymore for this function to allow the options to be unset. The provider type is always passed in in practice, and it seems better to require that going forward too. That also lets us simplify this function.

Instead of taking "args" and splitting it into options and callback by length, we can always expect options and a callback. This makes the types and JSDoc descriptions simpler too.

Reply:
Agreed! I haven't made this change yet, so it still needs to be done.